src/components/PadBtn/PadBtn.tsx

```diff
@@ -20,46 +20,71 @@ const PadBtn: React.FC<Props> = ({ input, calc, setCalc }) => {
       return;
     }
 
+    if (input === "<-") {
+      newCalc.displayedString = newCalc.displayedString.slice(0, -1);
+      getCalculation(newCalc);
+      setCalc(newCalc);
+      return;
+    }
+
     if (newCalc.displayedString.charAt(0) === "0")
       newCalc.displayedString = newCalc.displayedString.slice(0, -1); //clear first zero
 
-    let lastInputNum = newCalc.nums[newCalc.nums.length-1];
-    if (lastInputNum.length < 8)  newCalc.displayedString += input; //need improvement
+    let lastInputNum = newCalc.nums[newCalc.nums.length - 1];
```

Comment:
Since you are only using this variable in this if statement, you could add the boolean return directly to it. Like
let isLessThan8 = newCalc.nums[newCalc.nums.length - 1].length < 8

And then add it directly to the if
if (isLessThan8) ...

This makes your code more readable